### PR TITLE
Add io.github.dxx_rebirth.dxx-rebirth

### PR DESCRIPTION
**Pull request by fpiesche**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.
 This supersedes [PR 5443](https://github.com/flathub/flathub/pull/5443) and [5442](https://github.com/flathub/flathub/pull/5443) in favour of a single combined Flatpak (at about 4MB total size) for both branches of the DXX-Rebirth project.
As discussed there, this distribution combines the Descent 1 and 2 executables from DXX-Rebirth into one Flatpak with a single build and separate `.desktop` files, plus a simple zenity-based launcher to choose a default game to launch from the main `io.github.dxx_rebirth.dxx-rebirth.desktop` (as AppStream validation requires the presence of a `.desktop` file matching the app id).
I have also added local metainfo/icon files as there's no "combined" metadata/icon for a single package upstream, just separate ones for the Descent 1/2 variants of the project respectively.

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:** https://github.com/dxx-rebirth/dxx-rebirth/issues/691#issuecomment-2237823331

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: io.github.dxx_rebirth.dxx-rebirth.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,142 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>io.github.dxx_rebirth.dxx-rebirth</id>
+  <name>DXX-Rebirth</name>
+  <developer id="io.github.dxx_rebirth.dxx-rebirth.contributors">
```
Comment (bbhtt) [inline io.github.dxx_rebirth.dxx-rebirth.metainfo.xml:5]:
```suggestion
  <developer id="io.github.dxx_rebirth">
```

File: io.github.dxx_rebirth.dxx-rebirth.metainfo.xml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    <name>DXX-Rebirth Contributors</name>
+    <url>https://github.com/dxx-rebirth/dxx-rebirth/graphs/contributors</url>
+  </developer>
+  <launchable type="desktop-id">io.github.dxx_rebirth.dxx-rebirth.desktop</launchable>
+  <provides>
+    <id>io.github.dxx_rebirth.dxx-rebirth.d2x-rebirth.desktop</id>
+    <id>io.github.dxx_rebirth.dxx-rebirth.d1x-rebirth.desktop</id>
+  </provides>
+  <icon type="stock">io.github.dxx_rebirth.dxx-rebirth</icon>
```
Comment (bbhtt) [inline io.github.dxx_rebirth.dxx-rebirth.metainfo.xml:14]:
```suggestion
```

File: io.github.dxx_rebirth.dxx-rebirth.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  <summary>Modern cross-platform port of the Descent engine</summary>
+  <metadata_license>CC0-1.0</metadata_license>
+  <project_license>
+    LicenseRef-proprietary=https://raw.githubusercontent.com/dxx-rebirth/dxx-rebirth/master/COPYING.txt,GPL-3.0
```
Comment (bbhtt) [inline io.github.dxx_rebirth.dxx-rebirth.metainfo.xml:18]: Does this pass validation? I think you need to use AND here.

File: io.github.dxx_rebirth.dxx-rebirth.metainfo.xml [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    <keyword>sourceport</keyword>
+    <keyword>port</keyword>
+  </keywords>
+  <releases>
+    <release version="0.60.0-beta2" date="2018-04-15"/>
+  </releases>
+  <content_rating type="oars-1.1">
+    <content_attribute id="violence-fantasy">intense</content_attribute>
+  </content_rating>
+  <update_contact>https://github.com/dxx-rebirth/dxx-rebirth/</update_contact>
```
Comment (bbhtt) [inline io.github.dxx_rebirth.dxx-rebirth.metainfo.xml:141]: this should be a email

**Comment by bbhtt** [comment]
Can you submit metainfo and icon upstream? Upstream doesn't ship any icons?

**Comment by fpiesche** [comment]
Upstream does include icons and metainfo (which I had previously PRed for the initial separate packages I'd submitted), but only for the Descent 1/2 branches of the engine separately. I've [submitted a PR](https://github.com/dxx-rebirth/dxx-rebirth/pull/799) to add a combined icon and metainfo file.

> this should be a email

There's no contact email for the project - is this an optional field?

**Comment by bbhtt** [comment]
Yea that's optional

**Comment by bbhtt** [comment]
bot, build io.github.dxx_rebirth.dxx-rebirth

**Comment by bbhtt** [comment]
where is the game data supposed to come from? And how do you return to choose descent / descent II dialogue after choosing one of them in the first time i.e how will someone switch to descent II ?

**Comment by fpiesche** [comment]
The build includes separate shortcuts for launching Descent and Descent 2 directly; the default game chooser is mostly for the purposes of giving the required `io.github.dxx_rebirth.dxx-rebirth.desktop` something to do. Would it be preferable to pop the selector dialog open every time when launching this?

**Comment by bbhtt** [comment]
I'd probably add a `--setup` or something argument that makes the chooser pop up again.

**Comment by fpiesche** [comment]
Given that there are separate desktop files for launching Descent 1/2 directly I've just removed the "default" mechanism, so running the unspecific dxx-rebirth shortcut will always pop open the dialog instead.

**Comment by bbhtt** [comment]
bot, build io.github.dxx_rebirth.dxx-rebirth

**Comment by fpiesche** [comment]
bot, build io.github.dxx_rebirth.dxx-rebirth

**Comment by fpiesche** [comment]
Broke the launcher script like a total buffoon there. Fixed that now, and build tested and working!

**Pull request closed without merging**